README.md

```diff
@@ -1,17 +1,152 @@
-# YamlDecoder
+# YamlDecoder :goggles:
 
-This is a simple python package to decode yaml files into dataclasses.
+This is a simple python package to decode yaml files into dataclasses with type check.
 
-This is similar to the way we do stuff in Go in the rest of our repositories.
+This works similarly to the [yaml decoder](https://pkg.go.dev/gopkg.in/yaml.v2) package in Go.
 
-## Getting Started
+## Using This Module
 
 Run this commmand in any python project you are working on (as long as it uses pipenv):
 
 ```bash
+# for a specific commit version
 pipenv install -e git+https://github.com/sportsball-ai/py-yaml-decoder.git@<commit-id>d#egg=yamldecoder
+# for the latest version
+pipenv install -e git+https://github.com/sportsball-ai/py-yaml-decoderd#egg=yamldecoder
```

Comment:
@dskart do we want `d#egg=`, or just `#egg=`?

Reply:
good catch I will make a new pr